ompi/mca/coll/basic/coll_basic_gatherv.c

```diff
@@ -143,17 +142,22 @@ mca_coll_basic_gatherv_inter(const void *sbuf, int scount,
             return OMPI_ERROR;
         }
 
+        reqs = coll_base_comm_get_reqs(module->base_data, size);
         for (i = 0; i < size; ++i) {
             ptmp = ((char *) rbuf) + (extent * disps[i]);
             err = MCA_PML_CALL(irecv(ptmp, rcounts[i], rdtype, i,
                                      MCA_COLL_BASE_TAG_GATHERV,
                                      comm, &reqs[i]));
             if (OMPI_SUCCESS != err) {
+                ompi_coll_base_free_reqs(reqs, size);
```

Comment:
@bosilca Replace size by i+1: no need to try freeing all the allocated requests since only i+1 of them have been filled.